graylog2-server/src/main/java/org/graylog/plugins/sidecar/rest/resources/ConfigurationVariableResource.java

```diff
@@ -0,0 +1,199 @@
+import org.graylog.plugins.sidecar.services.ConfigurationService;
+import org.graylog.plugins.sidecar.services.ConfigurationVariableService;
+import org.graylog.plugins.sidecar.services.EtagService;
+import org.graylog2.audit.jersey.AuditEvent;
+import org.graylog2.audit.jersey.NoAuditEvent;
+import org.graylog2.plugin.rest.PluginRestResource;
+import org.graylog2.plugin.rest.ValidationResult;
+import org.graylog2.shared.rest.resources.RestResource;
+import org.mongojack.DBQuery;
+
+import javax.inject.Inject;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.List;
+import java.util.regex.Pattern;
+
+@Api(value = "Sidecar/ConfigurationVariables", description = "Manage collector configuration variables")
+@Path("/sidecar/configuration_variables")
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+@RequiresAuthentication
+public class ConfigurationVariableResource extends RestResource implements PluginRestResource {
+    private final ConfigurationVariableService configurationVariableService;
+    private final ConfigurationService configurationService;
+    private final EtagService etagService;
+
+    @Inject
+    public ConfigurationVariableResource(ConfigurationVariableService configurationVariableService, ConfigurationService configurationService, EtagService etagService) {
+        this.configurationVariableService = configurationVariableService;
+        this.configurationService = configurationService;
+        this.etagService = etagService;
+    }
+
+    @GET
+    @RequiresPermissions(SidecarRestPermissions.CONFIGURATIONS_READ)
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "List all configuration variables")
+    public List<ConfigurationVariable> listConfigurationVariables() {
+        final List<ConfigurationVariable> configurationVariables = this.configurationVariableService.all();
+
+        return configurationVariables;
+    }
+
+    @GET
+    @Path("/{id}/configurations")
+    @RequiresPermissions(SidecarRestPermissions.CONFIGURATIONS_READ)
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Show configurations using this variable")
+    public List<Configuration> getConfigurationVariablesConfigurations(@ApiParam(name = "id", required = true)
+                                                                       @PathParam("id") String id) {
+        final ConfigurationVariable configurationVariable = configurationVariableService.find(id);
+        if (configurationVariable == null) {
+            throw new NotFoundException("Could not find ConfigurationVariable <" + id + ">.");
```

Comment:
I would extract loading and checking a variable into a function, as we seem to be using it in some other places.